src/overlays/actors/ovl_Obj_Mine/z_obj_mine.c

```diff
@@ -24,131 +53,1182 @@ ActorInit Obj_Mine_InitVars = {
     sizeof(ObjMine),
     (ActorFunc)ObjMine_Init,
     (ActorFunc)ObjMine_Destroy,
-    (ActorFunc)ObjMine_Update,
-    (ActorFunc)ObjMine_Draw,
+    (ActorFunc)ObjMine_Path_Update,
+    (ActorFunc)ObjMine_Path_Draw,
 };
 
-// static ColliderJntSphElementInit sJntSphElementsInit[1] = {
-static ColliderJntSphElementInit D_80A84570[1] = {
+static ColliderJntSphElementInit sJntSphElementsInit[1] = {
     {
-        { ELEMTYPE_UNK2, { 0x00000000, 0x00, 0x00 }, { 0x01CBFFBE, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+        {
+            ELEMTYPE_UNK2,
+            { 0x00000000, 0x00, 0x00 },
+            { 0x01CBFFBE, 0x00, 0x00 },
+            TOUCH_NONE | TOUCH_SFX_NORMAL,
+            BUMP_ON,
+            OCELEM_ON,
+        },
         { 0, { { 0, 0, 0 }, 30 }, 100 },
     },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_80A84594 = {
-    { COLTYPE_METAL, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    ARRAY_COUNT(sJntSphElementsInit), D_80A84570, // sJntSphElementsInit,
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_METAL,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_1,
+        OC2_TYPE_1,
+        COLSHAPE_JNTSPH,
+    },
+    ARRAY_COUNT(sJntSphElementsInit),
+    sJntSphElementsInit,
+};
+
+static f32 sPathSpeeds[] = {
+    1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A845E8[] = {
+static ObjMineMtxF3 sStandardBasis = {
+    { 1.0f, 0.0f, 0.0f },
+    { 0.0f, 1.0f, 0.0f },
+    { 0.0f, 0.0f, 1.0f },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 1300, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 150, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_STOP),
 };
 
-#endif
+void ObjMine_Path_MoveToWaypoint(ObjMine* this, s32 index) {
+    Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->waypoints[index]);
+}
+
+s32 ObjMine_GetUnitVec3f(Vec3f* src, Vec3f* dst) {
+    f32 magnitude = Math3D_Vec3fMagnitude(src);
+
+    if (magnitude < 0.001f) {
+        return false;
+    }
+
+    dst->x = src->x * (1.0f / magnitude);
+    dst->y = src->y * (1.0f / magnitude);
+    dst->z = src->z * (1.0f / magnitude);
+    return true;
```

Comment:
```suggestion
    dst->z = src->z * (1.0f / magnitude);

    return true;
```
to match the other function.